src/gluonts/shell/env.py

```diff
@@ -29,18 +30,27 @@ def __init__(self, *args, **kwargs):
         sagemaker.TrainEnv.__init__(self, *args, **kwargs)
         self.datasets = self._load()
 
-    def _load(self) -> Dict[str, Dataset]:
+    def _load(self) -> Dict[str, Union[Dataset, Predictor]]:
         if "metadata" in self.channels:
             path = self.channels.pop("metadata")
             self.hyperparameters["freq"] = MetaData.parse_file(
                 path / "metadata.json"
             ).freq
 
+        model: Optional[Predictor] = None
+        if "model" in self.channels:
+            path = self.channels.pop("model")
+            with tarfile.open(path / "model.tar.gz") as targz:
```

Comment:
Have you tried running this?

But I guess we need to extract this ourselves, since SageMaker doesn't do it for us this time?

Reply:
I did, needed a couple of attempts :-) but this works